ontologytimemachine/custom_proxy.py

```diff
@@ -15,69 +16,69 @@
 IP = '0.0.0.0'
 PORT = '8899'
 
+config = ({'format': 'turtle', 'precedence': 'enforcedPriority', 'patchAcceptUpstream': False}, 'originalFailoverLiveLatest', False, 'all', False, True, None, None)
 
 logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
 logger = logging.getLogger(__name__)
 
 class OntologyTimeMachinePlugin(HttpProxyBasePlugin):
     def __init__(self, *args, **kwargs):
+        logger.info('Init')
         super().__init__(*args, **kwargs)
-        (self.ontoFormat, self.ontoVersion, self.only_ontologies,
-         self.https_intercept, self.inspect_redirects, self.forward_headers,
-         self.subject_binary_search_threshold) = parse_arguments()
-
+        (self.ontoFormat, self.ontoVersion, self.restrictedAccess,
+         self.httpsInterception, self.disableRemovingRedirects, 
+         self.forward_headers, self.timestamp, self.manifest) = config
```

Comment:
https://github.com/kuefmz/ontology-time-machine/commit/4c98e271eabc617e96b602761de53420bb297efb

Reply:
everything that is not a boolean but a string with a fixed set of choices is supposed to be an Enum.

Reply:
ontoVersion e.g. should not be a string.
httpsInterception is not a boolean ...
read my message and my code example careful again.
fixe add params: choices are supposed to be generated from the enums, default values from the default config 
the help must not include the choices or default values manually

Reply:
https://github.com/kuefmz/ontology-time-machine/commit/5f00b2771d694786beee4ec54768ff5dd119840d
https://github.com/kuefmz/ontology-time-machine/commit/2110f4db181c706d7279e9ae65de32b9d82265ca

Reply:
cool. but the enums are not used yet, 
see e.g. here in the if statements https://github.com/kuefmz/ontology-time-machine/blob/28c5b0878cad69fc0cbf5242729b51673ea018db/ontologytimemachine/utils/proxy_logic.py#L132 
maybe we should make an issue out of it?

Reply:
Maybe because I am not sure how I should used the Enums here. I used them in the parameter parsing
https://github.com/kuefmz/ontology-time-machine/blob/abeebc8ef25f35a5522cd5fe16235c8065f1d5c5/ontologytimemachine/utils/config.py#L72

Reply:
made an issue out of it 
its an easy task for copilot
https://chatgpt.com/share/6710c219-9abc-8006-a3ad-19cab0665892

Reply:
https://github.com/dbpedia/ontology-time-machine/issues/96